### PR TITLE
doc: add more context about security in the pipeline and operator

### LINKED ISSUE #4348: [Docs]: Missing some security checks we already do
### Is there an existing issue already for your request/idea?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new documentation enhancement to be evaluated.

### What problem in the existing documentation this issue aims to solve?

Some security checks and measures are not documented in the `docs/src/security.md` like:
Snyk process
SBOMs
Security bug reports
Security advisories
This needs to be reviewed and documented any missing process

### Describe what additions need to be done to the documentation

_No response_

### Describe what pages need to change in the documentation, if any

_No response_

### Describe what pages need to be removed from the documentation, if any

_No response_

### Additional context

_No response_

### Backport?

Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by sxd**
The current documentation doesn't cover all the security scans and measures taken in the last year, so we added that to the documentation and also a little section explaining how the operator permissions are managed in the different ways the operator supports and some recommendations about how to deploy it. In the new section, we also explain more about the cluster-wide permissions that are mandatory for the operator to work.

Closes #4348

### CODE REVIEW

File: docs/src/security.md [hunk 1]
```diff
@@ -367,3 +388,66 @@ For further detail on how `pg_ident.conf` is managed by the operator, see the
...
+and other permissions will be cluster wide.
+
+### Why cluster wide permissions are needed?
+
+Currently the operator requires cluster wide permissions only for `namespace` and
+`nodes` objects, all the other permissions can be namespaced or cluster wide.
+
+The reason for these permissions are:
+* namespace: required to watch other namespaces to keep track of the
+  the objects.
```
Comment (mnencia) [inline docs/src/security.md:435]:
```suggestion
* namespace: required to handle the deletion of namespaces containing Clusters.
```

File: docs/src/security.md [hunk 2]
```diff
@@ -367,3 +388,66 @@ For further detail on how `pg_ident.conf` is managed by the operator, see the
...
+* nodes: required to get the list of nodes and know the state of them so we avoid
+  scheduling any pod in a node that can't be used.
```
Comment (mnencia) [inline docs/src/security.md:437]:
```suggestion
* nodes: required to switch to a different instance if the primary is on a node being cordoned or drained. Without this, the PodDistructionBudget would keep the primary Pod from being evicted, preventing the maintenance operation from being completed.
```

File: docs/src/security.md [hunk 3]
```diff
@@ -26,6 +26,25 @@ a linter that scans the abstract syntactic tree of the source against a set of r
...
+and included in our CI/CD pipeline, thus, will block any pull request on which it could find
+any security issue. The configuration of CodeQL was set to only review Go code, so any other
+language in the repository like, python or bash, is not covered by this test.
+
+The [Snyk](https://snyk.io/) has an Open Source Projects in which CloudNativePG was accepted and
+we run the Snyk code scan every night in a scheduled job and produce weekly reports with any
+new finding on code security and also for licensing issues that could come up.
+
+One of the last things to note is that we have the "Report a vulnerability" option enabled
+on the Security section inside the repository that allows users to safely report any security
```
Comment (jsilvela) [inline docs/src/security.md:44]: I don't get this. We do have documentation explaining how to report vulnerabilities.
Do you mean, that section? If so I'll link here.
Been looking at the repo on GH and I don't see any vulnerability reporting tooling.

File: docs/src/security.md [hunk 4]
```diff
@@ -367,3 +388,66 @@ For further detail on how `pg_ident.conf` is managed by the operator, see the
...
 data protection in production environments, we highly recommend that you choose
 a storage class that supports encryption at rest.
+
+## Cluster wide permissions and operator deployment
+
+### Using manifest
+
+In the default installation for a vanilla Kubernetes cluster, the operator is deployed
+using the manifest that is build using `Kustomize` on the yaml files generated
+by `controller-gen` using the decorations available on [Kubebuilder](https://book.kubebuilder.io/),
```
Comment (jsilvela) [inline docs/src/security.md:398]: "decorations" ...???  Do you mean declarations? Customizations? Or is declaration a part of the lingo, like Ansible "playbooks"?

File: docs/src/security.md [hunk 5]
```diff
@@ -367,3 +394,69 @@ For further detail on how `pg_ident.conf` is managed by the operator, see the
...
+## Cluster wide permissions and operator deployment
+
+### Using the manifest
+
+In the default installation for a vanilla Kubernetes cluster, the operator is
+deployed using the manifest built using `Kustomize` on the yaml files generated
+by `controller-gen` using the markers available on [Kubebuilder](https://book.kubebuilder.io/).
+Sadly, these is a bit limited for now, and we can create a manifest
+with all permissions for RoleBinding or ClusterBinding, in our case, the permissions
+are cluster wide in the default deployment, thus, we recommend to deploy the operator
```
Comment (jsilvela) [inline docs/src/security.md:407]: I don't understand what you mean here.

File: docs/src/security.md [hunk 6]
```diff
@@ -367,3 +394,69 @@ For further detail on how `pg_ident.conf` is managed by the operator, see the
...
+from [OperatorHub.io](https://operatorhub.io/operator/cloudnative-pg) and thus, we have
+a little bit more help on how we manage the permissions.
+
+The first things to understand it's that OLM allows the users to enable the operator
+to "watch" one, a few or all the namespaces depending on how it is deployed, this open
+the possibility to be more granular on the permissions.
+
+When using OLM the default set of permissions is set to be the two types needed,
+RoleBinding and ClusterBinding, meaning that a set of permissions will be namespaced
+and other permissions will be cluster wide.
```
Comment (jsilvela) [inline docs/src/security.md:432]: This is confusingly expressed.
Do you mean that by default, OLM restricts permissions as much as possible? I.e. namespaced where possible?

File: docs/src/security.md [hunk 7]
```diff
@@ -367,3 +394,69 @@ For further detail on how `pg_ident.conf` is managed by the operator, see the
...
+
+### Why are cluster wide permissions needed?
+
+Currently the operator requires cluster wide permissions only for `namespace`
+and `nodes` objects. All the other permissions can be namespaced or cluster
+wide.
+
+The reason for these mandated permissions are:
+
+- namespace: required to handle the deletion of namespaces containing Clusters.
```
Comment (tijeshjose001z6h) [inline docs/src/security.md:442]: The reason for the namespace doesn't seem justified. Could you please elaborate and provide more clarity on this topic, as it is critical from a security perspective?

**Comment by gbartolini** [comment]
I have temporarily removed the entire section about cluster-wide permissions. I am creating a separate ticket about it, as I think it contradicts - or at least it is not entirely aligned with - the "RBAC" section. I also think that, instead of being thrown at the bottom of the page, it should sit under or next to the RBAC section (which assumes we deploy with the manifest).

**Review by gbartolini — APPROVED** [review]

**Pull request merged**